listenbrainz/spotify_updater/spotify_read_listens.py

```diff
@@ -267,12 +264,23 @@ def process_one_user(user):
     if user.token_expired:
         try:
             user = spotify.refresh_user_token(user)
-        except spotify.SpotifyAPIError:
-            current_app.logger.error('Could not refresh user token from spotify', exc_info=True)
-            raise
+        except spotify.SpotifyAPIError as e:
+            # if it is an error from the Spotify API, show the error message to the user
+            spotify.update_last_updated(
+                user_id=user.user_id,
+                success=False,
+                error_message=str(e),
+            )
+            if not current_app.config['TESTING']:
+                notify_error(user.musicbrainz_row_id, str(e))
```

Comment:
I don't think it's a great idea to send the contents of an exception to a user. Do you know if these exception messages from spotipy are nice to read? Can a user get more detailed information if they read this error, or is the result of all of them just going to be "you should re-authenticate"?

Reply:
This will send "Could not authenticate with Spotify, please unlink and link your account again." to the user.